src/fhirpath.js

```diff
@@ -458,6 +458,18 @@ function makeParam(ctx, parentData, type, param) {
 
 function doInvoke(ctx, fnName, data, rawParams){
   var invoc = engine.invocationTable[fnName];
+  if (ctx?.vars?._fns?.[fnName[0]]){
```

Comment:
This could work as a solution to temporarily add/change functions just to debug them.
But I wouldn't use this in production as running this code on every function call in every evaluation will impact performance.
If we need this in production, we need something like this here:
`var invoc = ctx.userInvocationTable[fnName] || engine.invocationTable[fnName];`

As an idea, we can have a separate function (same as the `solveInternalTypes` function) to create a user-defined invocation table. We can then put the result of this function in `ctx.userDefinedInvocationTable`.
But if we automatically convert `userFn(param1, param2)` to `{fn: userFn, arity:{2: ['Any', 'Any']} }`, we will skip the ability to customize a different number of parameters and parameter types.

So I think it could be better to just put the user-defined invocation table in `ctx.userInvocationTable` and expect the full description of the parameters in that object.

P.S.: We currently allow to override `trace()` using `options.traceFn`. Adding `ctx.userInvocationTable` would be another way to do this.